test/Swashbuckle.AspNetCore.SwaggerGen.Test/SchemaGenerator/JsonSerializerSchemaGeneratorTests.cs

```diff
@@ -979,6 +979,57 @@ public void GenerateSchema_GeneratesSchema_IfParameterHasTypeConstraints()
             Assert.Equal("integer", schema.Type);
         }
 
+        [Theory]
+        [InlineData(typeof(GenericType<string>), "string")]
+        [InlineData(typeof(GenericType<int>), "number")]
+        public void GenerateSchema_GeneratesSchemaWithCorrectType_IfTypeTakenFromUnderlyingType(Type genericType, string expectedType)
+        {
+            var subject = Subject(
+                configureGenerator: c => c.CustomTypeMappings.Add(genericType, mappingContext =>
+                {
+                    Assert.Equal(genericType, mappingContext.UnderlyingType);
+                    Assert.NotNull(mappingContext.SchemaGenerator);
+                    Assert.NotNull(mappingContext.SchemaRepository);
+
+                    var genericTypeArgument = mappingContext.UnderlyingType.GenericTypeArguments.First();
+
+                    if (genericTypeArgument == typeof(string))
+                        return new OpenApiSchema { Type = "string" };
+                    if (genericTypeArgument == typeof(int))
+                        return new OpenApiSchema { Type = "number" };
+
+                    throw new NotImplementedException();
```

Comment:
It would be better to fail with an assertion related to what unexpected value was received.